lib/modules/service.js

```diff
@@ -109,7 +110,46 @@ class Service {
   }
 
   _buildResult(response, body) {
+    let err = false;
+    let msg = [];
+    const $ = cheerio.load(body);
+
+    // Not found, unauthorized
+    if (response.statusCode == 404 || response.statusCode == 401) {
+      err = true;
+      // Add and Get return errors in different formats because why not?
+      let message = $('.alert').text();
+      if (message == '') {
+        message = $('.error_message').text();
+      }
+
+      // History (and probably others) don't include error messages because of course.
+      if (message == '') {
+        message = 'Unknown error. Most likely the group does not exist.';
+      }
+
+      msg.push(message.trim());
+    }
+
+    // Invalid netids sent into the Add method are returned in a list in the html
+    let missing = $('.notfoundmembers').find($('.notfoundmember')).length;
```

Comment:
Couldn't `$('.notfoundmembers')` return a value that the subsequent `.find` on this line cause an exception?  Especially when `_buildResult` is used by everything?

Reply:
It looks like cheerio always returns an object when using the selector so this shouldn't ever happen. In the tests, `_buildResult` is already processing xhtml returned from gws that doesn't contain elements with those classes.

I did just do some additional testing to make sure and it doesn't throw any errors on the nested `.find` and it doesn't matter if the first selector finds anything or not. The `.find` is always called without error.